compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -556,7 +582,13 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
                 );
 
                 if borrowed_place_has_mut_interior {
-                    self.check_op(ops::CellBorrow);
+                    // Locals without StorageDead follow the "enclosing scope" rule, meaning
+                    // they are essentially anonymous static items themselves.
+                    if self.local_has_storage_dead(place.local) {
```

Comment:
Somewhere around here there should be a note [like this](https://github.com/rust-lang/rust/pull/80418#issuecomment-752125143).